pyseir/inference/infer_rt.py

```diff
@@ -965,7 +968,7 @@ def replace_outliers(
             slicer = slice(idx - local_lookback_window, idx + local_lookback_window)
             changed_snippets.append(x[slicer].astype(int).tolist())
             try:
-                x[idx] = np.mean([x[idx - 1], x[idx + 1]])
+                x[idx] = np.mean([x.iloc[idx - 1], x.iloc[idx + 1]])
```

Comment:
I think this way of fixing the problem makes the most sense. It was confusing numpy and pandas indexing, so I just made it clearer that we are using numpy based indexing, which will throw the correct error. The switch to flatnonzero is recommended in the docs when people are using where, but not changing the value immediately inline. Future work to simplify and use np.where correctly, but it's not a high priority and this works (without the magic call to [0] at the end either).

Reply:
Ah yeah that makes sense!